yarn-project/p2p/src/services/libp2p/libp2p_service.ts

```diff
@@ -178,7 +179,7 @@ export class LibP2PService<T extends P2PClientType> extends WithTracer implement
 
     const otelMetricsAdapter = new OtelMetricsAdapter(telemetry);
 
-    const node = await createLibp2p({
+    const node = (await createLibp2p({
```

Comment:
this sort of cast should basically never be needed - can this have a type parameter or otherwise retain more type information about your addition here? (e.g. usually pattern is to have first arg be T extends some base type so the type T can be used to inform the return type)

Reply:
You're completely correct, its not required at all, updating